features/ui/header/header.tsx

```diff
@@ -0,0 +1,105 @@
+import React from "react";
+import styled from "styled-components";
+import { Routes } from "@config/routes";
+import { HeaderItemLink } from "./header-item-link";
+import { breakpoint } from "@styles/theme";
+import {
+  CustomButton,
+  ButtonSize,
+  ButtonColor,
+} from "@features/ui/button/customButton";
+
+const headerItems = [
+  { text: "Home", href: Routes.home },
+  { text: "Products", href: Routes.products },
+  { text: "Documentation", href: Routes.documentation },
+  { text: "Pricing", href: Routes.pricing },
+];
+
+const HeaderWrapper = styled.header`
+  width: 100%;
+  height: 80px;
+  padding: 0 2rem;
+  box-sizing: border-box;
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  background: white;
+`;
+
+const List = styled.ul`
+  list-style: none;
+  display: flex;
+  padding: 0;
+  margin: 0;
+  @media (max-width: ${breakpoint("mobile")}) {
+    display: none;
+  }
+`;
+
+const LinkList = styled(List)`
+  flex: 1;
+  justify-content: center;
+`;
+
+const DashboardButton = styled(CustomButton)`
+  display: block;
+  @media screen (max-width: ${breakpoint("mobile")}) {
+    display: none;
+  }
+`;
+
+const HamburgerButton = styled(CustomButton)`
+  display: none;
+  @media screen (max-width: ${breakpoint("mobile")}) {
+    display: block;
+  }
+`;
+export const Header = () => {
+  // const [isMobile, setMobile] = useState(false);
```

Comment:
I tried to go with a useState and useEffect approach but I could find a variable in the header component for the useEffect to watch.